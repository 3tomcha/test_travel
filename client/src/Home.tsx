import React from 'react';
import { Row, Col } from 'react-bootstrap';
import Header from './components/Header';
import Searchform from './components/Searchform';

const Home = () => {
    return (
        <div>
            <Row>
                <Col><Header/></Col>
            </Row>
            <Row>
                <Col><Searchform/></Col>
            </Row>
            <Row>
                <Col></Col>
                <Col></Col>
            </Row>
        </div>
    );
};

export default Home;