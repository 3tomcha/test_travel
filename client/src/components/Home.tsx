import React, { useState } from 'react';
import { Row, Col } from 'react-bootstrap';
import Header from './Header';
import HotelList, { Coordinates } from './Hotellist';
import Searchform from './Searchform';


const Home = () => {
    const today = new Date();
    const tomorrow = new Date();
    tomorrow.setDate(tomorrow.getDate() + 1);
    const [startDate, setStartDate] = useState<Date | null>(today);
    const [endDate, setEndDate] = useState<Date | null>(tomorrow);
    const [coordinates, setCoordinates] = useState<Coordinates>({latitude: 35.68985755, longitude: 139.72353351});

    const handleStartDateChange: Function = (date: Date | null) => {
        setStartDate(date);
    }
    const handleEndDateChange: Function = (date: Date | null) => {
        setEndDate(date);
    }
    const handleAddressChange: Function = (coordinates: Coordinates) => {
        setCoordinates(coordinates);
    }

    return (
        <div>
            <Row>
                <Col><Header/></Col>
            </Row>
            <Row>
                <Col>
                    <Searchform 
                        checkinDate={startDate}
                        checkoutDate={endDate}
                        handleCheckinDateChange={handleStartDateChange}
                        handleCheckoutDateChange={handleEndDateChange}
                        handleAddressChange={handleAddressChange}/>
                </Col>
                <Col>
                    <HotelList
                        coordinates={coordinates}
                        checkinDate={startDate}
                        checkoutDate={endDate}
                    />
                </Col>
            </Row>
        </div>
    );
};

export default Home;