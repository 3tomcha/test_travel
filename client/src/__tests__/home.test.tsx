import * as React from 'react';
import '../setupEnzyme';
import Home from '../components/Home';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

test('aaa', () => {
    const home = shallow(<Home/>);
    console.log(home.debug());
    // const tree = renderer.create(<Home/>).toJSON();
    
    // expect(tree).toMatchSnapshot();
});
