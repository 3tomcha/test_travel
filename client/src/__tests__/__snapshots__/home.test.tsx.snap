// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aaa 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <header>
        <img
          src=""
        />
      </header>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        <div
          className="card"
        >
          <form>
            <div
              className="card-header"
            >
              地名
            </div>
            <div
              className="card-body"
            >
              <div
                className="input-group"
              >
                <input
                  className="form-control"
                  onChange={[Function]}
                  placeholder="東京都千代田区丸の内1丁目"
                />
              </div>
            </div>
            <div
              className="card-header"
            >
              チェックイン日
            </div>
            <div
              className="card-body"
            >
              <div
                className="react-datepicker-wrapper"
              >
                <div
                  className="react-datepicker__input-container"
                >
                  <input
                    className=""
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    readOnly={false}
                    type="text"
                    value="05/23/2021"
                  />
                </div>
              </div>
            </div>
            <div
              className="card-header"
            >
              チェックアウト日
            </div>
            <div
              className="card-body"
            >
              <div
                className="react-datepicker-wrapper"
              >
                <div
                  className="react-datepicker__input-container"
                >
                  <input
                    className=""
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    readOnly={false}
                    type="text"
                    value="05/24/2021"
                  />
                </div>
              </div>
            </div>
            <button
              onClick={[Function]}
            >
              検索する
            </button>
          </form>
        </div>
      </div>
    </div>
    <div
      className="col"
    />
  </div>
</div>
`;
